### PR TITLE
fix: checkpoint finalized event

### PR DESCRIPTION
**Pull request by syuhei176**
### Tasks

* Deposited -> CheckpointFinalized
* property inputs are list of bytes

### CODE REVIEW

File: contracts/actions/deposit.ligo [hunk 1]
```diff
@@ -25,25 +25,30 @@ begin
   // create state_update
   const state_update: state_update = record
     property = record
-      predicate_address = ("tz1TGu6TN5GSez2ndXXeDX6LgUDvLzPLqgYV":address);//TODO: Ownership predicate address
-      inputs = map 0n -> "tz1OwnerN5GSez2ndXXeDX6LgUDvLzPLqgYV"; end;//TODO: sender address?
+      predicate_address = ("tz1TGu6TN5GSez2ndXXeDX6LgUDvLzPLqgYV":address); // TODO: Ownership predicate address
+      inputs = list encode_primitive(sender) end;
```
Comment (daiki44) [inline contracts/actions/deposit.ligo]: The input array is different. Is it okay?
https://github.com/cryptoeconomicslab/ovm-contracts/blob/master/contracts/DepositContract.sol#L75

Comment (syuhei176) [reply contracts/actions/deposit.ligo]: ohhh. LIGO's `state_update.property` is StateObject, so it's ok. However, StateUpdate should be also Property, and it has 4 inputs! I'll fix later.

Comment (syuhei176) [reply contracts/actions/deposit.ligo]: fixed

File: contracts/actions/deposit.ligo [hunk 2]
```diff
@@ -23,27 +24,33 @@ begin
...
 
   // create state_update
-  const state_update: state_update = record
-    property = record
-      predicate_address = ("tz1TGu6TN5GSez2ndXXeDX6LgUDvLzPLqgYV":address);//TODO: Ownership predicate address
-      inputs = map 0n -> "tz1OwnerN5GSez2ndXXeDX6LgUDvLzPLqgYV"; end;//TODO: sender address?
+  const state_update: property = record
+    // TODO: Injecting StateUpdate predicate address
+    predicate_address = ("tz1TGu6TN5GSez2ndXXeDX6LgUDvLzPLqgYV" : address);
+    inputs = list
```
Comment (daiki44) [inline contracts/actions/deposit.ligo]: The order does not matter?
```
bytes[] memory inputs = new bytes[](4);
inputs[0] = abi.encode(address(this));
inputs[1] = abi.encode(depositRange);
inputs[2] = abi.encodePacked(getLatestPlasmaBlockNumber());
inputs[3] = abi.encode(_initialState);
```

Comment (syuhei176) [reply contracts/actions/deposit.ligo]: It's wrong! fixed.

File: test/deposit.test.js [hunk 3]
```diff
@@ -9,24 +9,50 @@ describe('DepositContract', function() {
...
+            data: [
+              'CheckpointFinalizedEvent',
+              [
+                'tz1TGu6TN5GSez2ndXXeDX6LgUDvLzPLqgYV',
+                '0x611050bb088be105b193a250bef9c4bebec26be512e9a29787f315fa30b5748c',
+                {
+                  subrange: { start_: 0, end_: 1 },
+                  state_update: {
+                    predicate_address: 'tz1TGu6TN5GSez2ndXXeDX6LgUDvLzPLqgYV',
+                    inputs: [
```
Comment (daiki44) [inline test/deposit.test.js]: Sorry, how about the test case?

File: test/testdata/initial_storage [hunk 4]
```diff
@@ -25,16 +25,13 @@ record
...
           state_update = record
-            range = record
-              start_ = 1n;
-              end_ = 1n;
-            end;
-            property = record
-              predicate_address = ("tz1TGu6TN5GSez2ndXXeDX6LgUDvLzPLqgYV": address);
-              inputs = map 0n -> "hoge"; end;
+            predicate_address = ("tz1TGu6TN5GSez2ndXXeDX6LgUDvLzPLqgYV": address);
+            inputs = list
```
Comment (daiki44) [inline test/testdata/initial_storage]: Also here.

File: test/deposit.test.js (hunk 3 above)
Comment (syuhei176) [reply test/deposit.test.js]: fixed

File: test/deposit.test.js [hunk 5]
```diff
@@ -9,24 +9,62 @@ describe('DepositContract', function() {
...
+            ],
+            block_height: 0
+          }
+        ]
+      )
+
+      assert.deepEqual(
+        result.postState.branches['tz1TGu6TN5GSez2ndXXeDX6LgUDvLzPLqgYV']
+          .deposited_ranges,
+        []
```
Comment (syuhei176) [inline test/deposit.test.js]: I think extend_deposited_ranges doesn't work correctly because post state is wrong. I'll fix next PR.

**Pull request merged**